### PR TITLE
feat: show dbs and show tables api

### LINKED ISSUE #86: Add showtable&showdb interface in apiserver

### PR DESCRIPTION
**Pull request by altale**
#86

### CODE REVIEW

File: src/apiserver/api_server_impl.h [hunk 1]
```diff
@@ -63,6 +65,8 @@ class APIServerImpl : public APIServer {
  private:
     std::unique_ptr<sdk::SQLRouter> sql_router_;
     InterfaceProvider provider_;
+    // TODO(zhihao zhao): Since sql_router_ needs a general pointer, I cannot find a proper kind of small pointer to use here.
```
Comment (vagetablechicken) [inline src/apiserver/api_server_impl.h]: It's OK, just leave a comment like 'Not owned.'

File: src/apiserver/api_server_test.cc [hunk 2]
```diff
@@ -475,6 +475,52 @@ TEST_F(APIServerTest, procedure) {
     ASSERT_TRUE(env->cluster_remote->ExecuteDDL(env->db, "drop table trans;", &status));
 }
 
+TEST_F(APIServerTest, getDBs) {
+    const auto env = APIServerTestEnv::Instance();
+    brpc::Controller show_cntl;  // default is GET
+    show_cntl.http_request().uri() = "http://127.0.0.1:8010/dbs";
+    env->http_channel.CallMethod(NULL, &show_cntl, NULL, NULL, NULL);
+    ASSERT_FALSE(show_cntl.Failed()) << show_cntl.ErrorText();
+    LOG(INFO) << "get dbs: " << show_cntl.response_attachment();
```
Comment (vagetablechicken) [inline src/apiserver/api_server_test.cc]: How about add some dbs(some weird names) and check the results?

Comment (altale) [reply src/apiserver/api_server_test.cc]: Thanks, I will try it. But how to add more databases to APIServerTestEnv::Instance?

Comment (vagetablechicken) [reply src/apiserver/api_server_test.cc]: `env->cluster_remote->CreateDB/DropDB` can do that.

File: src/apiserver/api_server_impl.cc [hunk 3]
```diff
@@ -389,6 +391,75 @@ void APIServerImpl::RegisterGetSP() {
...
+        if (!ok) {
+            writer << err.Set(status.msg);
+            return;
+        }
+        writer.StartObject();
+        writer.Member("dbs");
+        writer.StartArray();
+        for (auto db : dbs) {
+            writer & db;
+        }
```
Comment (dl239) [inline src/apiserver/api_server_impl.cc]: set code and msg.

File: src/apiserver/api_server_impl.cc [hunk 4]
```diff
@@ -389,6 +391,75 @@ void APIServerImpl::RegisterGetSP() {
...
+            writer << err.Set("Invalid path");
+            return;
+        }
+        auto db = db_it->second;
+        auto table = table_it->second;
+        auto table_info = cluster_sdk_->GetTableInfo(db, table);
+        writer.StartObject();
+        writer.Member("code") & 0;
+        if (table_info == nullptr) {
+            LOG(INFO) << "Not found";
```
Comment (dl239) [inline src/apiserver/api_server_impl.cc]: Delete or update to DLOG

**Pull request merged**